### PR TITLE
Added missing null literal op

### LINKED ISSUE #1915: BUG: OmniSciDB/MapD aggregate with where param doesn't work
Similar error reported https://github.com/ibis-project/ibis/issues/1620

Solution:

Add `ops.NullLiteral: lambda *args: 'CAST(NULL AS FLOAT)'`

### PR DESCRIPTION
**Pull request by xmnlab**
Fix #1915

### CODE REVIEW

**Comment by xmnlab** [comment]
it breaks in some cases that uses bool https://github.com/omnisci/omniscidb/issues/375
for now I am skipping some tests for OmniSciDB that break for this issue.

**Comment by xmnlab** [comment]
@cpcloud it is done for review :) thanks!

File: ibis/tests/all/test_generic.py [hunk 1]
```diff
@@ -38,9 +38,21 @@ def test_fillna_nullif(backend, con, expr, expected):
 @pytest.mark.parametrize(
     ('expr', 'expected'),
     [
-        (ibis.coalesce(5, None, 4), 5),
-        (ibis.coalesce(ibis.NA, 4, ibis.NA), 4),
-        (ibis.coalesce(ibis.NA, ibis.NA, 3.14), 3.14),
+        pytest.param(
+            ibis.coalesce(5, None, 4),
+            5,
+            marks=pytest.mark.skip_backends((MapD,)),  # OmniSciDB issue #375
```
Comment (cpcloud) [inline ibis/tests/all/test_generic.py]: Should this be `xfail_backends` so that when this starts to pass we'll get a failing test?

Comment (xmnlab) [reply ibis/tests/all/test_generic.py]: sounds great! thanks!

File: ibis/tests/all/test_generic.py [hunk 2]
```diff
@@ -38,9 +38,21 @@ def test_fillna_nullif(backend, con, expr, expected):
 @pytest.mark.parametrize(
     ('expr', 'expected'),
     [
-        (ibis.coalesce(5, None, 4), 5),
-        (ibis.coalesce(ibis.NA, 4, ibis.NA), 4),
-        (ibis.coalesce(ibis.NA, ibis.NA, 3.14), 3.14),
+        pytest.param(
+            ibis.coalesce(5, None, 4),
+            5,
+            marks=pytest.mark.xfail_backends((MapD,)),  # OmniSciDB issue #375
```
Comment (xmnlab) [inline ibis/tests/all/test_generic.py]: @cpcloud for some reason  `xfail_backends` here is breaking the test just for python 3.5:

```
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/_pytest/main.py", line 213, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/_pytest/main.py", line 257, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/pluggy/hooks.py", line 289, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/pluggy/manager.py", line 87, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/pluggy/manager.py", line 81, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/xdist/remote.py", line 68, in pytest_runtestloop
INTERNALERROR>     self.run_one_test(torun)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/xdist/remote.py", line 85, in run_one_test
INTERNALERROR>     self.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/pluggy/hooks.py", line 289, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/pluggy/manager.py", line 87, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/pluggy/manager.py", line 81, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/_pytest/runner.py", line 72, in pytest_runtest_protocol
INTERNALERROR>     runtestprotocol(item, nextitem=nextitem)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/_pytest/runner.py", line 87, in runtestprotocol
INTERNALERROR>     reports.append(call_and_report(item, "call", log))
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/_pytest/runner.py", line 169, in call_and_report
INTERNALERROR>     report = hook.pytest_runtest_makereport(item=item, call=call)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/pluggy/hooks.py", line 289, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/pluggy/manager.py", line 87, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/pluggy/manager.py", line 81, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/pluggy/callers.py", line 203, in _multicall
INTERNALERROR>     gen.send(outcome)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/_pytest/skipping.py", line 123, in pytest_runtest_makereport
INTERNALERROR>     rep = outcome.get_result()
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/_pytest/runner.py", line 243, in pytest_runtest_makereport
INTERNALERROR>     return TestReport.from_item_and_call(item, call)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/_pytest/reports.py", line 361, in from_item_and_call
INTERNALERROR>     longrepr = item.repr_failure(excinfo)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/_pytest/python.py", line 835, in repr_failure
INTERNALERROR>     return self._repr_failure_py(excinfo, style=style)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/_pytest/nodes.py", line 280, in _repr_failure_py
INTERNALERROR>     truncate_locals=truncate_locals,
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/_pytest/_code/code.py", line 535, in getrepr
INTERNALERROR>     return fmt.repr_excinfo(self)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/_pytest/_code/code.py", line 765, in repr_excinfo
INTERNALERROR>     reprtraceback = self.repr_traceback(excinfo)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/_pytest/_code/code.py", line 715, in repr_traceback
INTERNALERROR>     reprentry = self.repr_traceback_entry(entry, einfo)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/_pytest/_code/code.py", line 675, in repr_traceback_entry
INTERNALERROR>     s = self.get_source(source, line_index, excinfo, short=short)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/_pytest/_code/code.py", line 617, in get_source
INTERNALERROR>     lines.extend(self.get_exconly(excinfo, indent=indent, markall=True))
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/_pytest/_code/code.py", line 624, in get_exconly
INTERNALERROR>     exlines = excinfo.exconly(tryshort=True).split("\n")
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/_pytest/_code/code.py", line 460, in exconly
INTERNALERROR>     lines = format_exception_only(self.type, self.value)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/traceback.py", line 136, in format_exception_only
INTERNALERROR>     return list(TracebackException(etype, value, None).format_exception_only())
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/traceback.py", line 463, in __init__
INTERNALERROR>     _seen=_seen)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/traceback.py", line 443, in __init__
INTERNALERROR>     and exc_value.__cause__ not in _seen):
INTERNALERROR> TypeError: unhashable type: 'TMapDException'
.....INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/_pytest/main.py", line 213, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/_pytest/main.py", line 257, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/pluggy/hooks.py", line 289, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/pluggy/manager.py", line 87, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/pluggy/manager.py", line 81, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/pluggy/callers.py", line 203, in _multicall
INTERNALERROR>     gen.send(outcome)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/xdist/dsession.py", line 115, in pytest_runtestloop
INTERNALERROR>     self.loop_once()
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/xdist/dsession.py", line 138, in loop_once
INTERNALERROR>     call(**kwargs)
INTERNALERROR>   File "/opt/conda/envs/ibis-env/lib/python3.5/site-packages/xdist/dsession.py", line 180, in worker_workerfinished
INTERNALERROR>     assert not crashitem, (crashitem, node)
INTERNALERROR> AssertionError: ('ibis/tests/all/test_generic.py::test_coalesce[MapD-expr0-5]', <WorkerController gw1>)
INTERNALERROR> assert not 'ibis/tests/all/test_generic.py::test_coalesce[MapD-expr0-5]'
```

I will skip the test again :(

Comment (xmnlab) [reply ibis/tests/all/test_generic.py]: I had the same problem locally with python 3.5 .. probably it is related to old version of pymapd on python 3.5 CI

**Comment by xmnlab** [comment]
@cpcloud it is done for a new review.

**Comment by anmyachev** [comment]
@xmnlab in 'projects' tab the PR has status 'in progress' but should has 'done', shouldn't it?

**Comment by xmnlab** [comment]
thanks @anmyachev for catching that up!

**Pull request merged**